### ISSUE TITLE
Cannot close a running program via CTRL+C shortcut

### ISSUE THREAD

**Issue by H4kt** [microsoft/terminal#7959]
# Environment

```none
Windows build number:  10.0.19041.450
Windows Terminal version (if applicable): 1.3.2651.0 
```

# Steps to reproduce
Run any long-running terminal program such as `ping <address> -t`
Press `CTRL+C` on your keyboard

# Expected behavior
Terminal window should stay opened, but ping process should be stopped

# Actual behavior
Terminal window closes

**Comment by skyline75489** [comment]
See #198, #6258

**Comment by jdebp** [comment]
I cannot reproduce this with version 1.3.2651.0.  The terminal window happily remains open.  You're likely going to need to provide more detailed steps for reproducing this.

```
PS C:\> ping 127.0.0.1 -t

Pinging 127.0.0.1 with 32 bytes of data:
Reply from 127.0.0.1: bytes=32 time<1ms TTL=128
Reply from 127.0.0.1: bytes=32 time<1ms TTL=128
Reply from 127.0.0.1: bytes=32 time<1ms TTL=128

Ping statistics for 127.0.0.1:
    Packets: Sent = 3, Received = 3, Lost = 0 (0% loss),
Approximate round trip times in milli-seconds:
    Minimum = 0ms, Maximum = 0ms, Average = 0ms
Control-C
PS C:\>
```

**Comment by zadjii-msft** [comment]
@H4kt could you share the profile you're using to run `ping`? it sounds like ping is exiting cleanly, and then the Terminal is automatically closing, because the client closed. Does this repro if you run `cmd.exe` as the commandline, then run `ping` from that prompt?

**Comment by H4kt** [comment]
Oh, silly me.
Yeah, the cmd is starting via gsudo to be elevated and that causes the problem.
But I'd like to know if there is another way to open an elevated cmd in terminal which will handle the CTRL+C input correctly?

**Comment by zadjii-msft** [comment]
Right now, no. I'd take a look at #632, #1032, #5000 et. al for why mixed elevation is hard. You could probably just get away with `"closeOnExit": "never"` in your settings though. That'll prevent the Terminal from auto-closing when the client application exits, even gracefully.

**Comment by DHowett** [comment]
If you're using `gsudo cmd` or `gsudo powershell`, and not just `gsudo ping`, _and_ <kbd>Ctrl+C</kbd> is terminating your session.... you're going to have to file a bug on gsudo for not handling ^C correctly.

**Comment by gerardog** [comment]
Hi @H4kt,
This was an issue in gsudo (gerardog/gsudo#53) and was fixed in v0.7.2, in early August. Please update gsudo.
